core/geometry/src/serialization/BGFBAccessors.ts

```diff
@@ -5,7 +5,7 @@
 /** @packageDocumentation
  * @module Serialization
  */
-import { flatbuffers } from "flatbuffers";
+import * as flatbuffers from "flatbuffers";
```

Comment:
@calebmshafer flatbuffer 2.x versions have different exports than 1.12 did, so this statement throws an error once I upgrade the version.

This file [was generated](https://github.com/iTwin/itwinjs-core/commit/6082ee8ef94f50edd44ac207e7a3bb64130c7c64) by the flatbuffers compiler from a schema file, so [I didn't want to edit anything inside](https://github.com/iTwin/itwinjs-core/pull/3130/files#diff-fd0540d54abc8879f56b73e04b7114d83d6a36a9eb74e5430d0dfa8c2bc7c99eR20)

Unfortunately, though, I get a compile error on [this line](https://github.com/iTwin/itwinjs-core/blob/359e9c44b0fced580bee8ce062af00b2291f8105/core/geometry/src/serialization/BGFBAccessors.ts#L6021) because the union function isn't generic in the 2.x typings (but it was in 1.12).

I can force it back to the correct type with an `as`, but again, I don't want to edit this file because it was generated.

@EarlinLutz, I noticed you generated this file originally, but I can't find the schema from which it was made. Is that still accessible anywhere?
I'm thinking the 2.x compiler might produce the result we're looking for.

Reply:
@dassaf4 has replaced Earlin as the primary maintainer for core-geometry.
Try [here](https://dev.azure.com/bentleycs/iModelTechnologies/_git/imodel02?path=/iModelCore/GeomLibs/serialization/src/FlatBuffer).

Reply:
Paul found it---the schema file is allcg.fbs. Earlin generated BGFBAccessors.ts using ./typescript/generateTypescript.cmd, which invokes a local (!) flatcv1.12.0.exe to output JavaScript that is in turn massaged with gema into Typescript.

I don't see this exe anywhere, but I do see imodel02\iModelCore\libsrc\flatbuffers\bin\beflatc.exe (and also src\PPBase\flatbuffers\bin\beflatc.exe in MicroStation), which may need to be upgraded too.

Reply:
@bbastings should be able to help you out with flatbuffers if needed.

Reply:
Adding @jffmarker because I vaguely remember him making an enhancement to the class methods produced for the flatbuffer schemas...I think it might be the "has_" methods?

Reply:
The other schema files are: ElementGraphics.fbs, TextString.fbs, and DgnFont.fbs from [here](https://dev.azure.com/bentleycs/iModelTechnologies/_git/imodel02?path=/iModelCore/DgnPlatform/DgnCore).
The .h files are generated by CompileElementGraphics.bat, CompileDgnFontFbs.bat, and TextStringFbsCompile.bat in same directory.

Reply:
Yup, I did make several enhancements to the C++ FB code, but I don't recall working on our JS version. Most changes related to adopting Bentley types or C++-specific features, but I did add `has_` utility methods so you could differentiate when the buffer explicitly wrote a default value vs. when it was missing:
```c++
#ifdef BENTLEY_MODIFICATIONS
    // Generate has methods of the form:
    // bool has_name() const { return CheckField(offset); }
    for (auto const& fieldP : struct_def.fields.vec)
        {
        auto& field = *fieldP;
        if (field.deprecated)
            continue;

        code += "  bool has_" + field.name + "() const";
        code += " { return CheckField(";
        code += NumToString(field.value.offset);
        code += "); }\n";
        }
#endif
```